lib/naver/searchad/api/core/base_service.rb

```diff
@@ -6,20 +14,65 @@ class BaseService
           include Logging
 
           attr_reader :url
-          attr_reader :path
+          attr_reader :base_path
+          attr_accessor :request_options
```

Comment:
don't know this hasn't finished yet :), out of review scope, will revisit later again